core/src/banking_stage/forwarder.rs

```diff
@@ -372,11 +388,52 @@ mod tests {
         }
     }
 
+    async fn check_all_received(
+        socket: UdpSocket,
+        expected_num_packets: usize,
+        expected_packet_size: usize,
+        expected_blockhash: &Hash,
+    ) {
+        let SpawnTestServerResult {
+            join_handle,
+            exit,
+            receiver,
+            server_address: _,
+            stats: _,
+        } = setup_quic_server_with_sockets(vec![socket], None, TestServerConfig::default());
+
+        let now = Instant::now();
+        let mut total_packets = 0;
+        while now.elapsed().as_secs() < 5 {
+            if let Ok(packets) = receiver.try_recv() {
+                total_packets += packets.len();
+                for packet in packets.iter() {
+                    assert_eq!(packet.meta().size, expected_packet_size);
+                    let tx: VersionedTransaction = packet.deserialize_slice(..).unwrap();
+                    assert_eq!(
+                        tx.get_recent_blockhash(),
+                        expected_blockhash,
+                        "Unexpected blockhash, tx: {tx:?}, expected blockhash: {expected_blockhash}."
+                    );
+                }
+            } else {
+                sleep(Duration::from_millis(100)).await;
+            }
+            if total_packets >= expected_num_packets {
```

Comment:
If `>` in positive case we will wait for 5sec before stopping. I think this check (I copy-pasted it from other tests) is to stop test earlier rather than to check if we receive more than expected packets. But with `>=` we in fact cannot detect if we received more packets than expected.